chainer/functions/evaluation/accuracy.py

```diff
@@ -25,22 +28,35 @@ def forward(self, inputs):
         xp = cuda.get_array_module(*inputs)
         y, t = inputs
         y = y.reshape(len(y), -1)  # flatten
-        pred = y.argmax(axis=1)
-        return xp.asarray((pred == t).mean(dtype='f')),
+
+        if self.ignore_label is not None:
+            mask = (t == self.ignore_label)
+            ignore_cnt = mask.sum()
+
+            # will always be true when the true label is ignore_label
+            pred = xp.where(mask, self.ignore_label, y.argmax(axis=1))
```

Comment:
Also, we would need [Advanced Indexing](http://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#advanced-indexing "Advanced Indexing")